contents/handbook/engineering/developing-locally.md

```diff
@@ -413,6 +413,12 @@ This command automatically turns any feature flag ending in `_EXPERIMENT` as a m
 
 Backend side flags are only evaluated locally, which requires the `POSTHOG_PERSONAL_API_KEY` env var to be set. Generate the key in [your user settings](http://localhost:8000/settings/user#personal-api-keys).
 
+## Extra: Debugging with VS Code
+
+The PostHog repository includes [VS Code launch options for debugging](https://github.com/PostHog/posthog/blob/master/.vscode/launch.json). Simply go to the `Run and Debug` tab in VS Code, select the desired service you want to debug, and run it. Once it starts up, you can set breakpoints and step through code to see exactly what is happening. There are also debug launch options for frontend and backend tests if you're dealing with a tricky test failure.
+
+> **Note:** If you're running most of your PostHog services locally with `./bin/start`, and only want to debug the backend for example, make sure to comment out that service from the [start script temporarily](https://github.com/PostHog/posthog/blob/master/bin/start#L22). 
```

Comment:
Curious why you'd want to do this? Performance? I'm mostly running PostHog with the compound launch configuration that starts all services.

Reply:
oh, this was on advice from me lol; I've run into issues when running the debugger alongside the service that was started via `./bin/start`, but never when I've stopped the service from the main script and then run it with the debugger.  Now that's all I do.

Reply:
Yup, that's kind of expected. What I mean is why use `./bin/start` at all instead of starting all services with this
<img width="304" alt="Screenshot 2024-11-14 at 22 14 47" src="https://github.com/user-attachments/assets/bdeac89d-2066-442b-8cf3-89b6de1c8817">

Reply:
oh lol I guess sometimes I don't want to run all services in the context of the debugger; I just want to debug the Django server or something.  But yeah, what you said works too!

Reply:
Got it! That makes sense

Reply:
I'll mention both options, thanks @thmsobrmlr !